### PR TITLE
[GPU] Enable fused MoE for trinity-mini AFMoE (sub-128 group _size + routed_scaling_factor)

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Description of the issue(symptom, root-cause, how it was resolved)
  - Symptom: Trinity-mini (afmoe) model fails to take the fused MoE path on GPU:
    - The `FuseMOE3GemmCompressed` transformation does not match the routing subgraph, so the model falls back to a non-fused MoE implementation
    - Even if the pattern is forced to match, the OpenCL kernel (`moe_3gemm_swiglu_mlp.cl`) hard-codes `FAKE_GROUP_SIZE = 128` and a `#if SUBGROUP_SIZE == 32` guard, producing wrong/garbage output for weight quantization `group_size = 64` (used by trinity-mini afmoe)
  - Root cause:
    - The afmoe routing branch has an additional `Multiply(routed_scaling_factor)` between `Divide` and `Slice`, which the existing pattern matcher does not allow
    - `FAKE_GROUP_SIZE` and the related inner-loop layout in `moe_3gemm_swiglu_mlp.cl` were tied to hardware sub-group width, not to the actual quantization group size
  - How it was resolved:
    - Transformation — Allow an` optional<Multiply>` between `Divide` and `Slice` in the sigmoid-bias routing pattern. Apply the absorbed scale as a post-op `Multiply` on the fused MoE output. Insert a `Convert` when the scale dtype differs from the fused output dtype
    - Kernel — Compute `FAKE_GROUP_SIZE = min(GATE_UP_GROUP_SIZE, DOWN_GROUP_SIZE, 128)` instead of hard-coding 128, and replace `#if SUBGROUP_SIZE == 32` with `#if ELEMS_PER_LANE == 4` `(ELEMS_PER_LANE = FAKE_GROUP_SIZE / SUBGROUP_SIZE)` at 6 sites so the inner-loop layout follows the actual quantization group size rather than the hardware sub-group width

#### The code and line that caused this issue (if it is not changed directly)
 -  `src/plugins/intel_gpu/src/plugin/transformations/fuse_moe_3gemm_compressed.cpp` — sigmoid-bias routing pattern (`Divide → Slice` chain)
 - `src/plugins/intel_gpu/src/graph/impls/ocl_v2/moe_3gemm_swiglu_mlp.cl` — `FAKE_GROUP_SIZE` definition and `#if SUBGROUP_SIZE == 32` guards

#### Reproduction step and snapshot (if applicable. Do not attach for customer model)
```
# Trinity-mini afmoe, gs=64, sigmoid-bias + routed_scaling_factor
$ OV_GPU_DYNAMIC_QUANTIZATION_THRESHOLD=1 \
  python genai/tools/llm_bench/benchmark.py \
    -d GPU \
    -m <trinity-mini OV_FP16-4BIT_DEFAULT path> \
    -p "What is Perl?" \
    --apply_chat_template --disable_prompt_permutation
```

#### Problematic graph
 - afmoe sigmoid-bias routing has the shape `Sigmoid → Add(bias) → ... → Divide → Multiply(routed_scaling_factor) → Slice → ...` whereas the existing matcher only allowed `Divide → Slice`. The extra `Multiply` blocks the match
 - After the fix, the matcher absorbs the `Multiply`, and the rewritten subgraph becomes `MOE3GemmFusedCompressed → Convert(scale) → Multiply(scale) → ...`(post-op).
image

#### Checklist
 - [x] Is it a proper fix? (not a workaround)
 - [x] Did you include test case for this fix, if necessary?
   - `tests/unit/transformations/fuse_moe_3gemm_compressed_test.cpp`: new `FuseMOE3GemmCompressed_SigmoidBias_RoutedScalingFactor` test verifying pattern match and post-op Multiply insertion against a reference graph.
   - `tests/unit/test_cases/moe_3gemm_gpu_test.cpp`: extended `random` / `symmetric` / `shared` / `shared_symmetric` instantiations with `group_size = 64` cases (trinity-mini afmoe shape).
 - [x] Did you review existing test that can be extended to cover this scenario? Which test did you review?
   - Reviewed `moe_3gemm_compressed_gpu_random` / `_symmetric_random` / `_shared_random` / `_shared_symmetric_random` — extended in place with the new `group_size = 64` parameter set rather than adding a new suite
   - Reviewed transformation tests for the existing sigmoid-bias routing pattern in `fuse_moe_3gemm_compressed_test.cpp` and added a sibling test that exercises the optional `Multiply(routed_scaling_factor)` branch.
 
### Tickets:
 - [CVS-185030](https://jira.devtools.intel.com/browse/CVS-185030)

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/transformations/fuse_moe_3gemm_compressed.cpp [hunk 1]
```diff
@@ -199,6 +203,36 @@ FuseMOE3GemmCompressed::FuseMOE3GemmCompressed() {
...
+            auto scaled_node = pattern_map.at(sig_norm_scaled).get_node_shared_ptr();
+            // Multiply has two inputs; the one that's not `sig_norm` is the scale.
+            ov::Output<ov::Node> scale;
+            auto sig_norm_out = pattern_map.at(sig_norm);
+            for (size_t i = 0; i < scaled_node->get_input_size(); ++i) {
+                if (scaled_node->input_value(i) != sig_norm_out) {
+                    scale = scaled_node->input_value(i);
+                    break;
+                }
+            }
```
Comment (v-Golubev) [inline src/plugins/intel_gpu/src/plugin/transformations/fuse_moe_3gemm_compressed.cpp]: Let's add `sig_norm_scale` node in the pattern and pass it as the 2nd `sig_norm_scaled` parameter: in this case, we will not need to manually handle multiply inputs checks

File: src/plugins/intel_gpu/tests/unit/transformations/fuse_moe_3gemm_compressed_test.cpp [hunk 2]
```diff
@@ -217,6 +223,162 @@ INSTANTIATE_TEST_SUITE_P(smoke,
                              ::testing::Values(false, true)),
                          FuseMOE3GemmCompressedTest::get_test_case_name);
 
+namespace {
+// Build a sigmoid+bias routing subgraph with an extra Multiply(routed_scaling_factor) inserted
+// between Divide(eps-normalized) and Slice — mirroring trinity-mini afmoe lowering.
+// Returns {unsqueeze_moe, topk_idx, scale_const}.
+std::tuple<ov::Output<ov::Node>, ov::Output<ov::Node>, std::shared_ptr<ov::op::v0::Constant>>
+build_sigmoid_bias_routing_subgraph_with_scale(const ov::Output<ov::Node>& routing_logits,
```
Comment (v-Golubev) [inline src/plugins/intel_gpu/tests/unit/transformations/fuse_moe_3gemm_compressed_test.cpp]: Can this test case be a part of `FuseMOE3GemmCompressedTest`? We could e.g. introduce a new `MoERoutingType`: in this case I suppose we will not need to duplicate a big part of MOE creation.

**Review by v-Golubev — COMMENTED** [review]
Regarding the original issue: **_"Input ... hasn't been found in primitive_ids map"_**: from my perspective, it is hard to understand what exactly went wrong by this message. It looks like we can add a dedicated `OPENVINO_THROW` to this code:
https://github.com/openvinotoolkit/openvino/blob/a5247d3e30ec70f8027a38d64aa46136ec39bc44/src/plugins/intel_gpu/src/plugin/ops/moe.cpp#L120-L124
and indicate there that `FuseMOE3GemmCompressed` transformation wasn't applied. What do you think?

**Comment by andrew-k-park** [comment]
> Regarding the original issue: **_"Input ... hasn't been found in primitive_ids map"_**: from my perspective, it is hard to understand what exactly went wrong by this message. It looks like we can add a dedicated `OPENVINO_THROW` to this code:
> 
> https://github.com/openvinotoolkit/openvino/blob/a5247d3e30ec70f8027a38d64aa46136ec39bc44/src/plugins/intel_gpu/src/plugin/ops/moe.cpp#L120-L124
> 
> 
> and indicate there that `FuseMOE3GemmCompressed` transformation wasn't applied. What do you think?

Agree - added an explicit `OPENVINO_THROW` in the `GEMM3_SWIGLU` branch so the real cause (`FuseMOE3GemmCompressed` didn't match) is reported instead of the cryptic **primitive_ids** error.

File: src/plugins/intel_gpu/src/plugin/transformations/fuse_moe_3gemm_compressed.cpp (hunk 1 above)
Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/plugin/transformations/fuse_moe_3gemm_compressed.cpp]: Done — added a named `sig_norm_scale` and the callback now reads it directly from `pattern_map`.

File: src/plugins/intel_gpu/tests/unit/transformations/fuse_moe_3gemm_compressed_test.cpp (hunk 2 above)
Comment (andrew-k-park) [reply src/plugins/intel_gpu/tests/unit/transformations/fuse_moe_3gemm_compressed_test.cpp]: Done — added a `with_routed_scale` flag to `FuseMOE3GemmCompressedTestParams` and removed the standalone test.

**Review by v-Golubev — APPROVED** [review]
Transformations part LGTM

File: src/plugins/intel_gpu/src/graph/impls/ocl_v2/moe_3gemm_swiglu_mlp.cl [hunk 3]
```diff
@@ -5,8 +5,31 @@
...
+#    else
+#        define FAKE_GROUP_SIZE 128
+#    endif
+#else
+#    define FAKE_GROUP_SIZE 128
+#endif
+
+// Number of K-elements each work-item handles per gk-iteration via the
+// intel_sub_group_block_read tile. Drives the inner-loop variant selection.
+#define ELEMS_PER_LANE (FAKE_GROUP_SIZE / SUBGROUP_SIZE)
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/src/graph/impls/ocl_v2/moe_3gemm_swiglu_mlp.cl:34]: https://github.com/openvinotoolkit/openvino/blob/8e84bff0a41beffa893ee40ef361d2440c78da13/src/plugins/intel_gpu/src/graph/impls/ocl_v2/moe/moe_3gemm_swiglu_opt.cpp#L427
As shown above, `SUBGROUP_SIZE` could be 16, and `ELEMS_PER_LANE` is `4` in this case. Please check if it's okay.

Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/graph/impls/ocl_v2/moe_3gemm_swiglu_mlp.cl:34]: Confirmed safe for the SUBGROUP_SIZE=16 + group_size=64 case: `intel_sub_group_block_read_us4` reads `4 × SUBGROUP_SIZE` halves per gk-iteration, which equals `FAKE_GROUP_SIZE` in both (SIMD16, group=64) and (SIMD32, group=128) — so the `#if ELEMS_PER_LANE == 4` branch is correct on both Xe-LPG and Xe2

File: src/plugins/intel_gpu/tests/unit/test_cases/moe_3gemm_gpu_test.cpp [hunk 4]
```diff
@@ -564,7 +564,11 @@ TEST_P(moe_3gemm_compressed_gpu_random, moe_accuracy_test_random) {
                           : ref.run_reference_softmax(hidden_states, routing_weights, w0_data, w1_data, w2_data);
     // SigmoidBias routing performs all routing math (sigmoid, bias, normalization) in f16 on the kernel side,
     // while the reference uses f32, leading to slightly larger numerical divergence than Softmax.
-    const float base_tolerance = routing_type == cldnn::MOE3GemmFusedCompressed::RoutingType::SIGMOID_BIAS ? 0.2f : 0.1f;
+    // Sub-128 weight quantization group sizes also accumulate more drift (more group transitions per K).
+    float base_tolerance = routing_type == cldnn::MOE3GemmFusedCompressed::RoutingType::SIGMOID_BIAS ? 0.2f : 0.1f;
+    if (config.group_size < 128) {
+        base_tolerance *= 3.0f;
+    }
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/tests/unit/test_cases/moe_3gemm_gpu_test.cpp]: Why do we need this? In my mind, we might expect better accuracy with smaller group_size.

Comment (andrew-k-park) [reply src/plugins/intel_gpu/tests/unit/test_cases/moe_3gemm_gpu_test.cpp]: drop unnecessary 3x tolerance

**Review by e-ddykim — APPROVED** [review]

**Review by ahnyoung-paul — APPROVED** [review]
LGTM
